gaphor/SysML/blocks/block.py

```diff
@@ -1,9 +1,154 @@
+            "show_parts", self.update_shapes
+        ).watch(
+            "show_references", self.update_shapes
+        ).watch(
+            "subject[NamedElement].name"
+        ).watch(
+            "subject[NamedElement].namespace.name"
+        ).watch(
+            "subject[Classifier].isAbstract", self.update_shapes
+        ).watch(
+            "subject[Class].ownedAttribute.aggregation", self.update_shapes
+        )
+        attribute_watches(self, "Block")
+        operation_watches(self, "Block")
+        stereotype_watches(self)
+
+    show_stereotypes: attribute[int] = attribute("show_stereotypes", int)
+
+    show_attributes: attribute[int] = attribute("show_attributes", int, default=False)
+
+    show_operations: attribute[int] = attribute("show_operations", int, default=False)
+
+    show_parts: attribute[int] = attribute("show_parts", int, default=False)
+
+    show_references: attribute[int] = attribute("show_references", int, default=False)
+
+    def update_shapes(self, event=None):
+        self.shape = Box(
+            Box(
+                Text(
+                    text=lambda: stereotypes_str(self.subject, ["block"]),
+                    style={"min-width": 0, "min-height": 0},
+                ),
+                EditableText(
+                    text=lambda: self.subject.name or "",
+                    width=lambda: self.width - 4,
+                    style={
+                        "font-weight": FontWeight.BOLD,
+                        "font-style": FontStyle.ITALIC
+                        if self.subject and self.subject.isAbstract
+                        else FontStyle.NORMAL,
+                    },
+                ),
+                Text(
+                    text=lambda: from_package_str(self),
+                    style={"font-size": 10, "min-width": 0, "min-height": 0},
+                ),
+                style={"padding": (12, 4, 12, 4)},
+            ),
+            *(
+                self.show_attributes
+                and self.subject
+                and [attributes_compartment(self.subject)]
+                or []
+            ),
+            *(
+                self.show_operations
+                and self.subject
+                and [operations_compartment(self.subject)]
+                or []
+            ),
+            *(
+                self.show_parts
+                and self.subject
+                and [
+                    self.block_compartment(
+                        gettext("parts"),
+                        lambda a: a.association and a.aggregation != "none",
+                    )
+                ]
+                or []
+            ),
+            *(
+                self.show_references
+                and self.subject
+                and [
+                    self.block_compartment(
+                        gettext("references"),
+                        lambda a: a.association and a.aggregation == "none",
```

Comment:
I think this should be `a.aggregation == "none" or a.aggregation == "composite"`

Reply:
And fixed :smile: